contrib/generate-struct.py

```diff
@@ -0,0 +1,385 @@
+#!/usr/bin/python3
+# pylint: disable=invalid-name,missing-docstring
+#
+# Copyright (C) 2022 Richard Hughes <richard@hughsie.com>
+#
+# SPDX-License-Identifier: LGPL-2.1+
+
+import os
+import sys
+import argparse
+
+from enum import StrEnum
```

Comment:
CI doesn't seem to like this.  Why not just use a dict?
```
    from enum import StrEnum
ImportError: cannot import name 'StrEnum' from 'enum' (/usr/lib/python3.8/enum.py)
```

Reply:
Ahh, StrEnum was only added in 3.11 -- will do something simpler.